ee/clickhouse/client.py

```diff
@@ -149,9 +149,8 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
             finally:
                 execution_time = perf_counter() - start_time
 
-                if not timeout_task.done:
-                    QUERY_TIMEOUT_THREAD.cancel(timeout_task)
-                    timing("clickhouse_sync_execution_time", execution_time * 1000.0, tags=tags)
+                QUERY_TIMEOUT_THREAD.cancel(timeout_task)
```

Comment:
Does this work as you'd expect?

I'd imagine cancelling between running (popitem) & before it being finished would error out (possible because the run is not inside a lock).

```
>>> from collections import OrderedDict
>>> res = OrderedDict()
>>> res['a'] = 1
>>> res['b'] = 2
>>> res.popitem(last=False)
('a', 1)
>>> del res['a']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'a' 
```